addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4689,6 +4689,9 @@ registry.sizing = SnippetOptionWidget.extend({
                 }, 0);
 
                 self.options.wysiwyg.odooEditor.automaticStepActive('resizing');
+                await new Promise(resolve => {
+                    self.trigger_up("snippet_option_update", { onSuccess: () => resolve() });
+                });
```

Comment:
What's that supposed to be doing? :grimacing: You make an event handler async, which has simply no effect other than being able to use `await` but you `await` something which is not considered by anything then. This code here would lead to the same result as `self.trigger_up("snippet_option_update", { onSuccess: () => {} });`

This should be better included in the main Mutex of the editor so that resizing + updating-the-ui-after-resizing are properly queue with other actions.

Reply:
> What's that supposed to be doing?

I got tangled up I guess. But yes, in the end I wanted to use `self.trigger_up("snippet_option_update", { onSuccess: () => {} });`.

> This should be better included in the main Mutex of the editor so that resizing + updating-the-ui-after-resizing are properly queue with other actions.

Not sure what you mean here. Maybe it's because I don't fully understand how the actions and the mutex interact. What about resizing should be better included in the mutex?

Reply:
@robinlej Everything in the editor options is queued in the main mutex at the SnippetsMenu level: if you click on 3 options A, B, and C very quickly, what should happen is A, updateUI, B, updateUI, C, updateUI, regardless of the (asynchronous) time it takes for the editor to perform each action and each UI update. The goal being to have a consistent behavior (no race condition with synchronous action being done at random places in between asynchronous ones) and have the UI update... being done when it can be updated.

Here, this apparently works by chance: we have an option (resizing columns) which does not go through any of the `_select` logic / `_updateUI` logic etc. Simply triggering `snippet_option_update` at the right place is a functional improvement but it's probably a new source of race condition: you ask for an UI update "immediately" (without waiting for the current queue to be dealt with) and you don't wait for the UI update to be done, meaning any other option can be performed "meanwhile" the UI update.

What should be done here is probably something like the drag & drop: locking the mutex when starting the drag/resizing, unlocking once you are done resizing, then ask for an update UI which has to be added in the queue as well. Hint: `snippet_edition_request` ?